src/metabase/driver/postgres.clj

```diff
@@ -209,6 +213,94 @@
 
 (def ^:private ^:dynamic *enum-types* nil)
 
+(def ^:private table-type-clauses
+  {"TABLE"              [:raw "c.relkind = 'r' AND n.nspname !~ '^pg_' AND n.nspname <> 'information_schema'"]
+   "PARTITIONED TABLE"  [:raw "c.relkind = 'p' AND n.nspname !~ '^pg_' AND n.nspname <> 'information_schema'"]
+   "VIEW"               [:raw "c.relkind = 'v' AND n.nspname <> 'pg_catalog' AND n.nspname <> 'information_schema'"]
+   "FOREIGN TABLE"      [:raw "c.relkind = 'f'"]
+   "MATERIALIZED VIEW"  [:raw "c.relkind = 'm'"]})
+
+(defn- get-tables-sql
+  [schemas tables types]
+  ;; Should track this implementation https://github.com/davecramer/pgjdbc/blob/a714bfd/pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java#L1272
+  (sql/format
+   (cond-> {:select    [[:n.nspname :schema]
+                        [:c.relname :name]
+                        [[:case-expr [:or [(keyword "~") :n.nspname "^pg_"] [:= :n.nspname "information_schema"]]
```

Comment:
You could use this `or` clause should be used to filter out system tables, because we don't want their info anyway.

Reply:
did that in https://github.com/metabase/metabase/pull/39670/commits/b5b9b8a6f7cb305afe85b25e251889d441a6a838